_sources/iteration/mixed_up_code.rst

```diff
@@ -0,0 +1,319 @@
+Mixed Up Code Exercises
+-----------------------
+
+Answer the following **Mixed-Up Code** questions to practice what you have learned in this chapter.
+
+.. parsonsprob:: iteration_mixed_up_code_1
+    :numbered: left
+    :adaptive:
+    :noindent:
+    :practice: T
+
+    The program below should print out the even numbers between 20 and 40, inclusive,
+    but the code is mixed up and contains extra blocks. Put the necessary blocks
+    in the correct order.
+    -----
+    int main() {
+    =====
+       int n = 20;
+    =====
+       int n = 0; #distractor
+    =====
+       while (n <= 40) {
+    =====
+       while (n < 40) {                        #paired
+    =====
+          cout << n << endl;
+    =====
+          n = n + 2;
+    =====
+          n++;                 #distractor
+    =====
+          n = n * 2;                 #distractor
+    =====
+       }
+    =====
+    }
+
+.. parsonsprob:: iteration_mixed_up_code_2
+    :numbered: left
+    :adaptive:
+    :noindent:
+    :practice: T
+
+    The program below should count down from 100 to 0 in decrements of
+    10 but the code is mixed up and contains extra blocks. Put the necessary blocks
+    in the correct order.
+    -----
+    int main() {
+    =====
+       int n = 100;
+    =====
+       int n = 10; #distractor
+    =====
+       while (n >= 0) {
+    =====
+       while (n < 0) { #distractor
+    =====
+       while (n > 0) { #distractor
+    =====
+          cout << n << endl;
+    =====
+          n -= 10;
+    =====
+          n += 10;                 #distractor
+    =====
+       }
+    =====
+    }
+
+.. parsonsprob:: iteration_mixed_up_code_3
+    :numbered: left
+    :adaptive:
+    :noindent:
+    :practice: T
+
+    The program below should find the sum of the first 10 natural numbers,
+    but the code is mixed up and contains extra blocks. Put the necessary blocks
+    in the correct order.
+    -----
+    int main() {
+    =====
+       int n = 10; #distractor
+    =====
+       int sum = 0;
+    =====
+       int sum = n;  #distractor
+    =====
+       for (int n = 1; n <= 10; ++n) {
+    =====
+       for (int n = 1; n < 10; ++n) {  #paired
+    =====
+       while (n < 100) {  #distractor
+    =====
+       while (n <= 9) {  #distractor
+    =====
+          cout << n << endl;  #distractor
+    =====
+          sum = sum + n;
+    =====
+       }
+    =====
+        cout << sum << endl;
```

Comment:
I think this line `cout << sum << endl;` might be at the wrong indentation level